documentation/src/main/asciidoc/ch12.asciidoc

```diff
@@ -855,3 +855,52 @@ should be used as a default way of resolving the name (see how it is used in <<e
 - `org.hibernate.validator.spi.nodenameprovider.JavaBeanProperty` is an interface that holds metadata about a bean property. It
 extends `org.hibernate.validator.spi.nodenameprovider.Property` and provide some additional methods like `Class<?> getDeclaringClass()`
 which returns the class that is the owner of the property.
+
+[[section-fail-fast-on-property-violation]]
+=== Fail fast on property violation mode
+
+Fail fast on property violation mode allows to perform class level constraint validation only if all property level constrints have passed without any violations.
+Enabling such mode should make writing class level constraint validators easier and cleaner, as the user will be working with valid properties.
+Hance there will be no need in any additional `null` checks or data sanity checks.
```

Comment:
```suggestion
Hence there will be no need in any additional `null` checks or data sanity checks.
```

Reply:
thanks for noticing the typo :wink: updated the text and rebased the branch as it was quite outdated